Apptainer.def

```diff
@@ -0,0 +1,58 @@
+# This builds a dockerfile containing a working copy of PySR
+# with all pre-requisites installed.
+
+Bootstrap: docker
+From: rockylinux/rockylinux:{{ ROCKY_VERSION }}
+
+%arguments
+	JULIA_RELEASE=1.10
+	JULIA_RELEASE_VERSION=4
+	PYSR_VERSION=0.19.3
+	PYTHON_VERSION=3.12
+	ROCKY_VERSION=8
+
+%files
+	./requirements.txt /opt/pysr/requirements.txt
+	./pyproject.toml /opt/pysr/pyproject.toml
+	./setup.py /opt/pysr/setup.py
+	./pysr /opt/pysr/pysr
+
+%environment
+	# Where to find Julia and its libraries
+	export LD_LIBRARY_PATH="/opt/julia/lib:/opt/julia/lib/julia:$LD_LIBRARY_PATH"
+	export PATH="/opt/julia/bin:$PATH"
+	# Use the container Julia binary, put the Project TOML file in the users home directory
+	export PYTHON_JULIAPKG_EXE=/opt/julia/bin/julia
+	export PYTHON_JULIAPKG_PROJECT="$HOME/.pysr"
+
+%post
+	dnf install -y wget python{{ PYTHON_VERSION }} python{{ PYTHON_VERSION }}-pip
+
+	cd /opt
+
+	# Download the version of Julia specified by the JULIA_RELEASE and JULIA_RELEASE_VERSION arguments
+	JULIA=julia-{{ JULIA_RELEASE }}.{{ JULIA_RELEASE_VERSION }}
+
+	wget https://julialang-s3.julialang.org/bin/linux/x64/{{ JULIA_RELEASE }}/${JULIA}-linux-x86_64.tar.gz
+
+	# Install it in /opt/julia
+	tar zxf ${JULIA}-linux-x86_64.tar.gz
+	rm ${JULIA}-linux-x86_64.tar.gz
+	mv ${JULIA} julia
+
+	PATH="/opt/julia/bin:${PATH}"
+
+	cd /opt/pysr
+
+	# Install IPython and other useful libraries:
+	pip3 install --no-cache-dir ipython matplotlib
+
+	pip3 install --no-cache-dir -r /opt/pysr/requirements.txt
+
+	# Install PySR:
+	pip3 install --no-cache-dir .
+
```

Comment:
```suggestion
        python -c "import pysr"
```
This triggers the installation of Julia libraries so should be done within the install step

Reply:
Done. Note that because the container file system is read-only at runtime all the Julia stuff gets installed again, in the users $HOME/.pysr/ project, at runtime.

Reply:
> read-only at runtime all the Julia stuff gets installed again

Hm, I don't think this shouldn't happen. It should just use the read-only environment. Maybe one of the juliacall environment variables is not set to the apptainer dir? https://juliapy.github.io/PythonCall.jl/stable/juliacall/#julia-config

Reply:
Also see https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_LOAD_PATH and https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_DEPOT_PATH

Reply:
I guess it's tricky because Julia will also have the precompilation cache which might change depending on the user's environment variables. So ideally that should get written to a writeable directory, while the packages and registry are frozen by the image.

Reply:
I've played around with this some and while you can get Julia to refer to the read-only environment using JULIA_LOAD_PATH and JULIA_DEPOT_PATH it is unhappy because the file system is read only. I'll go back and get the exact error and location in the code. I've solved this in other Apptainer contexts by using a writable overlay but I was hoping to be able to put the PySR container in a Google Artifact Registry repository so that it only had to be build once.

Reply:
I'm testing a JULIA_DEPOT_PATH solution that looks promising. I should have a new commit today or tomorrow.

Reply:
Fixed. Thanks for the JULIA_LOAD_PATH and JULIA_DEPOT_PATH pointers.